src/instructlab/utils.py

```diff
@@ -900,3 +914,105 @@ def list_models(
 def contains_argument(prefix: str, args: typing.Iterable[str]) -> bool:
     # Either --foo value or --foo=value
     return any(s == prefix or s.startswith(prefix + "=") for s in args)
+
+
+def get_model_arch(model_path: pathlib.Path) -> str:
+    """
+    Extract a given model's architecture from its config if available and return it
+
+    args
+        model_path (str): Path to the model, used to read the model config if available
+    returns
+        model_arch (str): The architecture of the model as expressed in the config file
+    """
+
+    # TODO: implement for GGUF models
+
+    model_arch = "default"
+    try:
+        model_cfg = get_config_file_from_model(model_path, "config.json")
+        model_arch = model_cfg["model_type"]
+    except Exception as e:
+        logger.debug(
+            f"Unable to get model architecture from config for model: {model_path}: {e}. Falling back to default value"
+        )
+
+    return model_arch
+
+
+def get_config_file_from_model(model_path: pathlib.Path, filename: str):
+    """
+    Reads a config file from a model's directory into memory
+
+    args
+        model_path (str): path to the model
+        filename (str): name of config file to be read
+    returns
+        json loaded config dict
+    """
+    try:
+        with open(
+            pathlib.Path(model_path) / filename,
+            "r",
+            encoding="utf-8",
+        ) as f:
+            model_cfg = json.load(f)
+        return model_cfg
+    except (FileNotFoundError, NotADirectoryError, PermissionError) as e:
+        raise ValueError(
+            f"Unable to load file {filename} from {model_path}: {e}"
+        ) from e
+
+
+def get_model_template_from_tokenizer(model_path: pathlib.Path) -> Tuple[str, str, str]:
+    """
+    Attempts to read a model's tokenizer config file and extract the template and special
+    tokens from it
+
+    args
+        model_path (str): path to the model
+    returns
+        template (str): resolved chat template
+        bos_token (str): Beginning of sentence token
+        eos_token (str): End of sentence token
+    """
+    template = eos_token = bos_token = None
+
+    try:
+        tcfg = get_config_file_from_model(model_path, "tokenizer_config.json")
+        template = tcfg["template"]
```

Comment:
Head start with some simple unit tests as https://gist.github.com/bbrowning/eb4c2339f3426d05c874cfda9dcb753b that will catch this error, and also verifies get_model_arch doesn't error on granite-7b-lab's config. That's just an example - feel free to steal, ignore, or rewrite.

Reply:
ah, thanks for catching that! must've broken that in the shuffle since i last manually tested this with vllm

I was able to reproduce this myself, luckily chatting still worked because it fell back to the correct in memory template so that is good 

will add that unit test